cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocImageVersion.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.common;
+
+import io.cdap.cdap.runtime.spi.VersionInfo;
+import joptsimple.internal.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class used to compare versions from Dataproc cluster images
+ */
+public class DataprocImageVersion implements VersionInfo {
+  List<Integer> versionSegments;
+
+  public DataprocImageVersion(@Nullable String imageVersion) {
+    // For null versions, we assume zero.
+    if (imageVersion == null) {
+      imageVersion = "0";
+    }
+
+    versionSegments = new ArrayList<>();
+
+    // If the version contains a hyphen, only take the first portion (e.g. 1.2.3-debian4)
+    if (imageVersion.contains("-")) {
+      imageVersion = imageVersion.split("-")[0];
+    }
+
+    // As a safety, assume that missing elements default to zero. (e.g. 1.3. -> 1.3.0, 1..1 -> 1.0.1)
```

Comment:
is this what dataproc does? If it's not an allowed syntax, seems better to throw an exception